### ISSUE TITLE
@charset "UTF-8"; and "string-quotes": "single"

### ISSUE THREAD

**Issue by hugomrdias** [stylelint/stylelint#2895]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

bug

> Which rule, if any, is this issue related to?

"string-quotes": "single"

> What CSS is needed to reproduce this issue?

e.g.

```css
@charset "UTF-8";
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "string-quotes": "single"
  }
}
```

> Which version of stylelint are you using?

8.1.1

> How are you running stylelint: CLI, PostCSS plugin, Node API?

all

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no

> What did you expect to happen?

`@charset "UTF-8";` should be ignored for `"string-quotes": "single"`   

related links:    
[spec](https://www.w3.org/TR/css-syntax-3/#determine-the-fallback-encoding)
[prettier behavior](https://github.com/prettier/prettier/issues/2817#issuecomment-329452090)

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
[stylelint] Expected single quotes (string-quotes)
```

> stylelint should also enforce @chartset at the top of the stylesheet (or at least the file been processed)

**Comment by jeddy3** [comment]
@hugomrdias Thanks for the report and for using the template.

I think this is a dupe of https://github.com/stylelint/stylelint/issues/2788. That issue has "help wanted" label, but it's not been picked up yet.

**Comment by jeddy3** [comment]
> stylelint should also enforce @chartset at the top of the stylesheet (or at least the file been processed)

This sounds like a new rule e.g. `at-charset-no-invalid-position`. Feel free to create a new and separate issue for that.
